torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -137,14 +135,11 @@ ncclRedOpRAII getNcclReduceOp(
 #ifdef ENABLE_NCCL_PREMUL_SUM_SUPPORT
       switch (dataType) {
         case ncclHalf:
-          return unpackPreMulSum<at::Half, ncclHalf>(
-              reduceOp, comm, dev_in_group);
+          return unpackPreMulSum<at::Half, ncclHalf>(reduceOp, comm);
```

Comment:
hmm whats unpackPreMulSum

Reply:
PreMulSum is a ncclReduceOp. Compared to regular sum reduce, it multiplies each operand with a scalar of user's choice before summing them. For use in case user worries about gradient overflow. 

unpackPreMulSum here unpacks that scalar value out, from the composite ncclReduceOp datatype.